### ISSUE TITLE
Remove use of types module constants in twisted/spread

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8457]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)| @rodrigc reported|
|-|-|
|Trac ID|trac#8457|
|Type|enhancement|
|Created|2016-06-12 04:37:35Z|

The *types* module constants have been removed in Python 3.
For example using *types.TupleType* in Python 3 will give an *AttributeError*

These constants should be removed and replaced with the primitive type instead.  See:

http://www.diveintopython3.net/porting-code-to-python-3-with-2to3.html#types

<details><summary>Searchable metadata</summary>

```
trac-id__8457 8457
type__enhancement enhancement
reporter__rodrigc rodrigc
priority__normal normal
milestone__Python_3_x Python-3.x
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1465706255769673 1465706255769673
changetime__1474198049187875 1474198049187875
version__None None
owner__rodrigc rodrigc

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_1"></a>|@rodrigc commented|
|-|-|

See:
https://github.com/twisted/twisted/pull/177

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_2"></a>|@adiroiban set owner to @rodrigc|
|-|-|

I don't understand this PR :(

There is also a conditional py3 declaration for which I don't see any test.

Can you please describe what was changed here, what was broken and how it is now fixed?

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="markrwilliams's avatar" src="https://avatars.githubusercontent.com/u/704247?s=50" width="50" height="50">](https://github.com/markrwilliams)<a name="note_3"></a>|@markrwilliams commented|
|-|-|

Replying to [adiroiban](#note_2):
> I don't understand this PR :(

`twisted.spread.jelly` and `twisted.spread.pb` fail to import on Python 3 because the `types` module now lacks the expected primitive `*Type*` classes:

```
Python 3.5.2 (default, Jul  2 2016, 07:15:59) 
[GCC 4.2.1 Compatible FreeBSD Clang 3.4.1 (tags/RELEASE_34/dot1-final 208032)] on freebsd10
Type "help", "copyright", "credits" or "license" for more information.
>>> import twisted.spread.jelly
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/mrw/src/python/twisted/twisted/spread/jelly.py", line 70, in <module>
    from types import StringType
ImportError: cannot import name 'StringType'
>>> import twisted.spread.pb
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/mrw/src/python/twisted/twisted/spread/pb.py", line 46, in <module>
    from twisted.spread.jelly import jelly, unjelly, globalSecurity
  File "/home/mrw/src/python/twisted/twisted/spread/jelly.py", line 70, in <module>
    from types import StringType
ImportError: cannot import name 'StringType'
>>>
```

For these modules to be importable the code needs to be changed to check against primitive types themselves (e.g., `int`) except in the case of `FunctionType`, `LambdaType`, and a few others (see https://docs.python.org/3/library/types.html#types.FunctionType).

> There is also a conditional py3 declaration for which I don't see any test.

This is one step towards making this module importable at all under Python 3; after this, the tests should run under Python 3 and exercise the Python 3 specific code.  Is this not the way Twisted usually approaches minor API differences between 2 and 3?

Hope that all helps!

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_4"></a>|GitHub set status to `closed`|
|-|-|

In changeset 1230f84611c055d76e1b19f0afa096809ca3bbea
```
#!CommitTicketReference repository="" revision="1230f84611c055d76e1b19f0afa096809ca3bbea"
Merge pull request #177 from rodrigc/8457-rodrigc-types

Author: rodrigc
Reviewer: markrwilliams
Fixes: #8457

Replace use of types constants with primitive types in twisted/spread
```

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_5"></a>|@hawkowl commented|
|-|-|

[mass edit] Removing review from closed tickets.
